cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -0,0 +1,332 @@
+                writer.println("consumer.sasl.jaas.config=" + jaasConfig);
+                writer.println(consumerOauthCallbackClass);
+
+                writer.println("admin.sasl.mechanism=" + saslMechanism);
+                writer.println("admin.sasl.jaas.config=" + jaasConfig);
+                writer.println(adminOauthCallbackClass);
+
+                writer.println();
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Adds consumer client rack {@code rack.id}. The rack ID will be set in the container based on the value of the
+     * rack.id file (if it exists). This file is generated by the init-container used when rack awareness is enabled.
+     *
+     * @return Returns the builder instance
+     */
+    public KafkaConnectConfigurationBuilder withRackId()   {
+        printSectionHeader("Additional information");
+        writer.println("consumer.client.rack=${strimzidir:/opt/kafka/init:rack.id}");
+        writer.println();
+        return this;
+    }
+
+    /**
+     * Configures the Kafka Connect configuration providers
+     *
+     * @param userConfig    the user configuration to extract the possible user-provided config provider configuration from it
+     */
+    private void printConfigProviders(AbstractConfiguration userConfig) {
+        printSectionHeader("Config providers");
+        String strimziConfigProviders = "strimzienv,strimzifile,strimzidir";
+
+        if (userConfig != null && !userConfig.getConfiguration().isEmpty() && userConfig.getConfigOption("config.providers") != null) {
+            writer.println("# Configuration providers configured by the user and by Strimzi");
+            writer.println("config.providers=" + userConfig.getConfigOption("config.providers") + "," + strimziConfigProviders);
+            userConfig.removeConfigOption("config.providers");
+        } else {
+            writer.println("# Configuration providers configured by Strimzi");
+            writer.println("config.providers=" + strimziConfigProviders);
+        }
+
+        writer.println("config.providers.strimzienv.class=org.apache.kafka.common.config.provider.EnvVarConfigProvider");
+        writer.println("config.providers.strimzienv.param.allowlist.pattern=.*");
+        writer.println("config.providers.strimzifile.class=org.apache.kafka.common.config.provider.FileConfigProvider");
+        writer.println("config.providers.strimzidir.class=org.apache.kafka.common.config.provider.DirectoryConfigProvider");
+        writer.println("config.providers.strimzidir.param.allowed.paths=/opt/kafka");
+        writer.println();
+    }
+
+    /**
+     * Adds user provided Kafka Connect configurations.
+     *
+     * @param configurations   User provided Kafka Connect configurations
+     *
+     * @return Returns the builder instance
+     */
+    public KafkaConnectConfigurationBuilder withConfigurations(AbstractConfiguration configurations) {
```

Comment:
Maybe we can use `withUserConfiguration` as we have for the broker configuration builder?

Reply:
thanks Paolo! I renamed the method.